src/steamship/agents/functional/output_parser.py

```diff
@@ -0,0 +1,89 @@
+import json
+import re
+from typing import Dict, List, Optional
+
+from steamship import Block, MimeTypes, Steamship
+from steamship.agents.schema import Action, AgentContext, FinishAction, OutputParser, Tool
+from steamship.data.tags.tag_constants import RoleTag
+
+
+# TODO(dougreid): extract shared bits from this and the ReACT output parser into a utility?
+class FunctionsBasedOutputParser(OutputParser):
+
+    tools_lookup_dict: Optional[Dict[str, Tool]] = None
+
+    def __init__(self, **kwargs):
+        tools_lookup_dict = {tool.name: tool for tool in kwargs.pop("tools", None)}
+        super().__init__(tools_lookup_dict=tools_lookup_dict, **kwargs)
+
+    def _extract_action_from_function_call(self, text: str, context: AgentContext) -> Action:
+        wrapper = json.loads(text)
+        fc = wrapper.get("function_call")
+        name = fc.get("name", "")
+        tool = self.tools_lookup_dict.get(name, None)
+        if tool is None:
+            raise RuntimeError(
+                f"Could not find tool from function call: `{name}`. Known tools: {self.tools_lookup_dict.keys()}"
+            )
+
+        input_blocks = []
+        arguments = fc.get("arguments", "")
+        args = json.loads(arguments)
+        # TODO(dougreid): validation and error handling?
+
+        if args.get("text", None):
```

Comment:
```suggestion
        if text := args.get("text"):
```

1. We can use walrus operator to avoid 2x .get 
2. .get defaults to None

Reply:
walrus operator is available for 3.8+ which is the minimum Python version for our SDK.